challenges-services/src/main/java/org/exoplatform/challenges/listener/AnnouncementActivityGenerator.java

```diff
@@ -24,36 +27,42 @@
 import java.util.Map;
 
 public class AnnouncementActivityGenerator extends Listener<AnnouncementStorage, Announcement> {
-  private static final Log LOG = ExoLogger.getLogger(AnnouncementActivityGenerator.class);
+  private static final Log   LOG                        = ExoLogger.getLogger(AnnouncementActivityGenerator.class);
 
   public static final String ANNOUNCEMENT_ACTIVITY_TYPE = "challenges-announcement";
 
-  private ExoContainer container;
+  private ExoContainer       container;
 
-  private ActivityStorage activityStorage;
+  private ActivityStorage    activityStorage;
 
-  public AnnouncementActivityGenerator(ActivityStorage activityStorage, ExoContainer container) {
+  private ChallengeService   challengeService;
+
+  public AnnouncementActivityGenerator(ActivityStorage activityStorage,
+                                       ExoContainer container,
+                                       ChallengeService challengeService) {
     this.activityStorage = activityStorage;
+    this.challengeService = challengeService;
     this.container = container;
   }
 
   @Override
-  public void onEvent(Event<AnnouncementStorage, Announcement> event) throws ObjectNotFoundException {
+  public void onEvent(Event<AnnouncementStorage, Announcement> event) throws ObjectNotFoundException, IllegalAccessException {
     ExoContainerContext.setCurrentContainer(container);
     RequestLifeCycle.begin(container);
     try {
       Announcement announcement = event.getData();
       AnnouncementStorage announcementStorage = event.getSource();
-      ExoSocialActivity activity = createActivity(announcement);
+      Challenge challenge = challengeService.getChallengeById(announcement.getChallengeId(), Utils.getCurrentUser());
+      ExoSocialActivity activity = createActivity(announcement, challenge);
       announcement.setActivityId(Long.parseLong(activity.getId()));
-      announcementStorage.saveAnnouncement(announcement);
+      AnnouncementEntity announcementEntity = EntityMapper.toEntity(announcement, challenge);
+      announcementStorage.saveAnnouncement(announcementEntity);
```

Comment:
AnnouncementStorage should get an Announcement DTO then it converts it to an entity